doc/guides/dts/index.rst

```diff
@@ -16,18 +16,21 @@ PowerPC architecture.  However, it is now in use for ARM and other
 architectures.
 
 The device tree is a data structure for dynamically describing hardware
-using a Device Tree Source (DTS) data structure language, and compiled
-into a compact Device Tree Blob (DTB) using a Device Tree Compiler (DTC).
-Rather than hard coding every detail of a board's hardware into the
+using a Device Tree Source (DTS) data structure language. In Linux, it is
+compiled into a compact Device Tree Blob (DTB) using a Device Tree Compiler
+(DTC). Rather than hard coding every detail of a board's hardware into the
 operating system, the hardware-describing DTB is passed to the operating
 system at boot time. This allows the same compiled Linux kernel to support
 different hardware configurations within an architecture family (e.g., ARM,
 x86, PowerPC) and moves a significant part of the hardware description out of
 the kernel binary itself.
 
 Traditional usage of device tree involves storing of the Device Tree Blob.
-The DTB is then used during runtime for configuration of device drivers.  In
-Zephyr, the DTS information will be used only during compile time.
+The DTB is then used during runtime for configuration of device drivers.
+However, storing the DTB and the code required to parse it requires an amount
+of footprint that cannot be afforded on micro-controllers systems with limited
+resources that are the primary target of Zephyr.
+Hence, in Zephyr, the DTS information will be used only during compile time.
 Information about the system is extracted from the compiled DTS and used to
 create the application image.
```

Comment:
When I read this it sounds like two paragraphs of information that then abruptly ends with; "but we don't use this, so none of what you read is relevant."

Can we cut down on how much we talk about DTB. Since DTB is not in use, one would think we would be able to document our DT usage without even mentioning it.

Reply:
FWIW, I like the DTB stuff. Gives some context when looking up device tree info elsewhere.

Reply:
> Can we cut down on how much we talk about DTB?

I mostly agree, though it could be interesting at least explaining why we're not using dtb the usual way and process dts at target compile time. This is not obvious to every one, I already had to explain this a couple of times to people used to Linux and that can't figure out the memory constraints we're facing.
So this is more targeting Linux users than non Linux aware developers, but still a useful piece of information. Anyway, we can try to shorten it a bit.